### DISCUSSION TITLE
Track Elevation Data

### DISCUSSION THREAD

**Discussion by Alia1999** [theOehrly/Fast-F1#593] [Q&A]
Hi guys, 

Is there any way to get track coordinate data. I wanted to build a visualization to show track conditions, especially elevation differences. 
I assumed the lap.telemetery['Z'] refered to the elevation of the track and tried to to visualize it:
```

#additional calculations for scaling

t = np.linspace(0, 2, 620)
x = (lap.telemetry['X']/182000)
y = (lap.telemetry['Y']/182000)
z = (lap.telemetry['Z']/182000)*t
thicc = (lap.telemetry['Z']/70)*t+20 # from 0.5 to 2.5
    


# see https://stackoverflow.com/a/58880037/2749397 for explanation
segments = np.stack((np.c_[x[:-1],x[1:]],np.c_[y[:-1],y[1:]]), axis=2)
lc = LineCollection(segments, linewidths=thicc, color='grey')

fig, ax = plt.subplots(figsize=(10, 10))

ax.set_aspect(1)
ax.add_collection(lc)
ax.plot((lap.telemetry['X']/182000), (lap.telemetry['Y']/182000)+0.001 ,
        color='black', linestyle='-', linewidth=3, zorder=5)
ax.axis('off')
ax.autoscale()

plt.show()
```

and this was the result:
![image](https://github.com/theOehrly/Fast-F1/assets/76116384/dd7f2461-e453-4370-95e5-2a9f1f8518ff)

I can see quite a few differences in what it actually supposed to reflect: 

![image](https://github.com/theOehrly/Fast-F1/assets/76116384/9330e07f-cfd1-4884-8dcd-7d0539dda341)

Is the data I'm looking at wrong or is it my method/code? Would appreciate any insights on this

**Comment by theOehrly** [comment]
Yes, Z represents the elevation. Your code is wrong. You are visualizing the track from straight above (top view) in a 2D plot! You need to use a 3D plot in matplotlib and make your line segments acutally point along the z-axis. But you don't have a z-axis in this type of plot. There may be other problems as well.
